src/lib/utils.js

```diff
@@ -46,26 +47,129 @@ function isPermissionReadOnly(perm) {
   )
 }
 
+/**
+ * Finds the permission block for the the file
+ * in the permissions owned by the current cozy-client.
+ *
+ * Iterates through parent folders if needed
+ * until we can find the permissions attached to the share
+ *
+ * @private
+ * @param {string} fileId
+ * @param {string|undefined} parentId
+ */
+async function findPermissionFor(fileId, { parentId, client, permissions }) {
+  const perm = permissions.find(perm => {
+    !perm.values || perm.values.includes(fileId)
+  })
+  if (perm) {
+    return perm
+  } else if (parentId) {
+    const parentData = await getFile(client, parentId)
+    const parentParentId = parentData.attributes.dir_id
+    const context = { parentId: parentParentId, client, permissions }
+    return findPermissionFor(parentId, context)
+  } else {
+    return undefined
+  }
+}
+
+/**
+ * Get permissions on io.cozy.files for current context
+ * @private
+ * @param {CozyClient} client
+ * @returns {object}
+ */
+async function getFilesOwnPermissions(client) {
+  const permissionsData = await client
+    .collection('io.cozy.permissions')
+    .getOwnPermissions()
+  const permissionObject = get(permissionsData, 'data.attributes.permissions')
+  if (!permissionObject) throw `can't get self permissions`
+  const permissions = Object.values(permissionObject)
+  return permissions.filter(perm => perm.type === 'io.cozy.files')
+}
```

Comment:
should be moved to cozyclient

Reply:
yep, as  we discussed in a previous changes this week. Let me do that.

Reply:
https://github.com/cozy/cozy-client/pull/636

I will update this PR once the linked one lands

Reply:
done, we now use cozy-client/models/permission and cozy-client/models/file